src/internal/connector/onedrive/collections.go

```diff
@@ -14,94 +14,89 @@ import (
 	"github.com/alcionai/corso/src/internal/observe"
 	"github.com/alcionai/corso/src/pkg/logger"
 	"github.com/alcionai/corso/src/pkg/path"
-	"github.com/alcionai/corso/src/pkg/selectors"
 )
 
-// Collections is used to retrieve OneDrive data for a
-// specified user
+type driveSource int
+
+const (
+	unknownDriveSource = iota
+	OneDriveSource
+	SharePointSource
+)
+
+type isAnyMatcher interface {
```

Comment:
this is a bit nitpicky, but `isAnyMatcher` seems like a bit of a misnomer to me. It makes me think the interface only implements `IsAny()`

Reply:
maybe just `folderMatcher` or `scopeMatcher` ?

Reply:
`folderMatcher` sounds good, since we're packing the folder category into the match assumption.